index.d.ts

```diff
@@ -0,0 +1,168 @@
+import Eris from "eris";
+import EventEmitter from "events"
+
+type Callback = (...args: any[]) => void;
+
+interface ShardStats {
+  id: number;
+  ready: boolean;
+  latency: number;
+  status: "disconnected" | "connecting" | "handshaking" | "ready" | "resuming";
+}
+
+interface ClusterStatsInfo {
+  guilds: number;
+  users: number;
+  totalRam: number;
+  voice: number;
+  exclusiveGuilds: number;
+  largeGuilds: number;
+  clusters: ClusterDataStats[];
+}
+
+interface ClusterDataStats extends Exclude<ClusterStatsInfo, "totalRam" | "clusters"> {
+  cluster: number;
+  shards: number;
+  ram: number;
+  uptime: number;
+  shardStats: ShardStats;
+}
+
+interface ClusterStats {
+  stats: ClusterStatsInfo;
+  clustersCounted: number;
+}
+
+interface ClusterData {
+  item: number;
+  value: {
+    id: number;
+    clusterCount: number;
+    name: string;
+    firstShardID: number;
+    lastShardID: number;
+    maxShards: number;
+    token: string;
+    file: string;
+    clientOptions: Eris.ClientOptions;
+  }
+}
+
+interface Webhook {
+  id: string;
+  token: string;
+}
+
+interface Webhooks {
+  shard?: Webhook;
+  cluster?: Webhook;
+}
+
+interface ClusterManagerOptions {
+  shards?: number;
+  firstShardID?: number;
+  lastShardID?: number;
+  clusters?: number;
+  clusterTimeout?: number;
+  stats?: boolean;
+  debug?: boolean;
+  statsInterval?: number;
+  name?: string;
+  guildsPerShard?: number;
+  webhooks?: Webhooks;
+  clientOptions: Eris.ClientOptions;
+}
+
+declare class IPC extends EventEmitter {
+  events: Map<string, { fn: Callback }>;
+  register(event: string, callback: Callback): void;
+  unregister(name: string): void;
+  broadcast(name: string, message?: any): void;
+  sendTo(cluster: number, name: string, message?: any): void;
+  fetchUser(id: string): Promise<Eris.User>;
+  fetchGuild(id: string): Promise<Eris.Guild>;
+  fetchChannel(id: string): Promise<Eris.AnyChannel>;
+  fetchMember(guildID: string, memberID: string): Promise<Eris.Member>;
+}
+
+declare class Cluster<T extends Base, C extends Eris.Client> {
+  shards: number;
+  maxShards: number;
+  firstShardID: number;
+  lastShardID: number;
+  mainFile: string | null;
+  clusterID: number;
+  clusterCount: number;
+  guilds: number;
+  users: number;
+  uptime: number;
+  exclusiveGuilds: number;
+  largeGuilds: number;
+  voiceChannels: number;
+  shardsStats: ShardStats[];
+  app: T | null;
+  bot: C;
+  test: boolean;
+  ipc: IPC;
+  logOverride(message: unknown): string;
+  spawn(): void;
+  connect(firstShardID: number, lastShardID: number, maxShards: number, token: string, type: unknown, clientOptions: Eris.ClientOptions): void;
```

Comment:
Type here should be `connect | reconnect`. However, I have just realized now that type is probably a legacy parameter that can probably be removed considering it no longer serves a purpose.

Reply:
To be honest that can be its own PR, don't want to include it here